pkg/test/utils.go

```diff
@@ -140,3 +143,63 @@ func IgnoreLeadingComments(s string) string {
 	}
 	return strings.TrimSpace(strings.Join(out, "\n")) + "\n"
 }
+
+func CompareGoldenObject(path string, got []byte) error {
+	writeGoldenOutput := os.Getenv("WRITE_GOLDEN_OUTPUT") != ""
+	want, err := os.ReadFile(path)
+	if err != nil {
+		if writeGoldenOutput && os.IsNotExist(err) {
+			// Expected when creating output for the first time
+			if err := os.WriteFile(path, got, 0644); err != nil {
+				return fmt.Errorf("failed to write golden output %s: %w", path, err)
+			}
+			fmt.Printf("wrote updated golden output to %s", path)
+			return nil
+		} else {
+			return fmt.Errorf("failed to read golden file %q: %w", path, err)
+		}
+	}
+	var wantMap, gotMap map[string]interface{}
+	err = yaml.Unmarshal(want, &wantMap)
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(got, &gotMap)
+	if err != nil {
+		return err
+	}
+	if ok, err := compareNestedFields(wantMap, gotMap); !ok {
+		return err
+	}
+	return nil
+}
+
+func compareNestedFields(wantMap, gotMap map[string]interface{}) (bool, error) {
```

Comment:
You can probably use cmp.Diff for this, but ... is the output any better than using cmp.Diff on the file (and then `git diff` if we write the file)?

Reply:
Thanks, I took a look at cmp.Diff method. I think what I’m trying to do here is to verify the existence of fields in the object; I don’t think we really care about the actual value in each field?

The values might be generated (like date or timestamp), we need to replace them with a fixed value for golden comparison. But for this case, I think we only want to verify that all fields are the same when compared to the golden file, please correct me if I’m wrong...

Reply:
I thought we do care both the fields the values? Eventually the replay tests will be applied to all the test cases to verify if the KCC logic is executed as expected or not, and being able to verify the status fields are more of a "side effect" that we can leverage immediately.